### ISSUE TITLE
Hello, request the missing code for model testing in the public githob project in the paper "PhAI: A deep-learning approach to solve the crystallographic phase problem" and several test datasets mentioned in the paper (since I only see the training and inference code you posted on githob, not your githob The part of the project that is used for code and test datasets for model testing, I would like to ask you to share it, thanks).

### ISSUE THREAD

**Issue by e823912** [aslarsen/PhAI#1]
hello,

I'm a, I'm currently an AI for Science enthusiast.I am honored to have read your paper in a "scientic" journal titled "PhAI: A deep-learning approach to solve the crystallographic phase problem" and have great admiration for your research work.

After reading your dissertation, I was encouraged and intrigued by your research methods and results. In order to gain a deeper understanding of your research methodology and analysis process, I would very much like to be able to test the model from the Githob code used in your paper and from the several test datasets mentioned in the paper.

I understand that these resources may involve intellectual property and confidentiality issues, but I guarantee that they will be used strictly in accordance with the relevant usage regulations, for academic research purposes only, and that you and your team will give appropriate academic credit in any citations or subsequent research based on these codes and data.

If possible, please let me know how to access these resources or if you need any additional information from me. I very much look forward to the opportunity to communicate with you further and thank you for your contributions to the academic community,thank you!

**Comment by aslarsen** [comment]
Hi, thank you for showing interest in our work. 
Checkout the published archive at https://erda.ku.dk/archives/0f226a473de37f517efb893f24868e10/published-archive.html It contains the parameters, training data and code to test the network. 

The CSD and ICSD databases are both licenced databases, even for academics and we can’t share that data. If you affiliated with an academic institution there is a high chance that someone has a license. The CSD also have an online portal for structure download https://www.ccdc.cam.ac.uk/structures/

The Crystallography Open Database(COD) also contain many structures that you can test on.

**Comment by e823912** [comment]
Hello, I've received your reply, thank you very much! However, why is the code of your PhAI_archive file not similar to the code of test.py model testing, I only see model training and model inference, not generally model training and then model testing, and finally model inference, I always feel that your code is missing this part of the code to test the model, and I have seen your training data and inference data in the dataset posted on that PhAI_archive file, Isn't that PhAI_archive file of yours the data of the CSD and ICSD databases, and isn't the test set data and the training data of the 2387 structures in the same database? Also, is the data preprocessing of the test data the same as the training data, because the training data is to be generated, and the 2387 structures of the test dataset are also composed of artificially generated crystal structures like the training set?

**Comment by e823912** [comment]
Dear Dr. Anders Støttrup Larsen,

Hello!

First of all, I would like to express my sincere gratitude for your previous replies regarding PhAI's data and source code. Your answer was very helpful to me.&nbsp;&nbsp; &nbsp;

Secondly, I also want to ask you why the code of your PhAI_archive file is not similar to the code of the test.py model test, I only see the model training and model inference, not generally the first model training and then the model test, and finally the model inference, I always feel that your code is missing the part of the code to test the model, and I have seen your training data and inference data in the dataset posted on that PhAI_archive file, Isn't that PhAI_archive file of yours the data of the CSD, ICSD database and COD database, and the test set data of the 2387 structures and the training data not in the same database? Also, is the data preprocessing of the test data the same as the training data, because the training data is to be generated, and the 2387 structures of the test dataset are also composed of artificially generated crystal structures like the training set?

Thank you for taking the time to read this email and look forward to hearing from you.

Wish: smooth work and happy life!



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "aslarsen/PhAI"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年10月7日(星期一) 凌晨3:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [aslarsen/PhAI] Hello, request the missing code for model testing in the public githob project in the paper "PhAI: A deep-learning approach to solve the crystallographic phase problem" and several test datasets mentioned in the paper (since I only see the training and inference code you posted on githob, not your githob The part of the project that is used for code and test datasets for model testing, I would like to ask you to share it, thanks). (Issue #1)





 
Hi, thank you for showing interest in our work.
 Checkout the published archive at https://erda.ku.dk/archives/0f226a473de37f517efb893f24868e10/published-archive.html It contains the parameters, training data and code to test the network.
 
The CSD and ICSD databases are both licenced databases, even for academics and we can’t share that data. If you affiliated with an academic institution there is a high chance that someone has a license. The CSD also have an online portal for structure download https://www.ccdc.cam.ac.uk/structures/
 
The Crystallography Open Database(COD) also contain many structures that you can test on.

**Comment by aslarsen** [comment]
In the Phai.zip the data/sg14_max10_fulldataset.zarr contains only the artificial generated structures. The CSD and ICSD data is not included in the .zip file. The COD structures and experimental amplitudes can be downloaded from the COD website. 
If you have experimental crystal then you can calculate the corresponding amplitudes and target phases using the function phasecalc_torch found in training_backend.py. The function calculate_F in the crystallography_module.py found in Phai.zip can also calculate the structure factors.

There are two types of test data. (1) Experimental determined crystal structures with calculated amplitudes and phases this is the case for CSD and ICSD and (2) experimentally measured amplitudes which is the case for the COD database. I recommend that you read “Section 5 Crystal structure data sets” in the supplementary information.

See the README.txt in the Phai.zip file PhAI_wrapper folder on how to make predictions using PhAI.

**Comment by e823912** [comment]
OK, I see, thank you very much for your detailed response to my previous question. The explanations and suggestions you provided were very useful and helped me better understand the more specific features of the source code and datasets, and I hope to ask you further if I have any other questions about the source code and datasets you provided, thank you!

**Comment by aslarsen** [comment]
Glad I could help.
